x/mongo/driver/topology/rtt_monitor_test.go

```diff
@@ -265,3 +270,88 @@ func TestMin(t *testing.T) {
 		})
 	}
 }
+
+func TestPercentile(t *testing.T) {
+	cases := []struct {
+		desc       string
+		samples    []time.Duration
+		minSamples int
+		percentile float64
+		want       time.Duration
+	}{
+		{
+			desc:       "Should return 0 for fewer than minSamples samples",
+			samples:    []time.Duration{1, 0, 0, 0},
+			minSamples: 2,
+			percentile: 90.0,
+			want:       0,
+		},
+		{
+			desc:       "Should return 0 for empty samples slice",
+			samples:    []time.Duration{},
+			minSamples: 0,
+			percentile: 90.0,
+			want:       0,
+		},
+		{
+			desc:       "Should return 0 for no valid samples",
+			samples:    []time.Duration{0, 0, 0},
+			minSamples: 0,
+			percentile: 90.0,
+			want:       0,
+		},
+		{
+			desc:       "First tertile when minSamples = 0",
```

Comment:
Did not know [tertile](https://en.wiktionary.org/wiki/tertile#:~:text=tertile%20(plural%20tertiles),the%20three%20groups%20so%20divided.) was a word 😆 .

Reply:
Tt's also referred to as a [tercile](https://en.wiktionary.org/wiki/tercile) lol.  The entire nomenclature is super peculiar.  The 2-quantile version of it is the "median" and then the "tiles` take over until the 8th-quantile and then it becomes the "ciles" (decile, hexadecile, ...).

Reply:
Sometimes it's spelled [tercile](https://en.wiktionary.org/wiki/tercile).  The whole nomenclature is really weird:

> [median](https://en.wiktionary.org/wiki/median#English) (2-quantile), [tercile](https://en.wiktionary.org/wiki/tercile)/tertile (3), [quartile](https://en.wiktionary.org/wiki/quartile) (4), [quintile](https://en.wiktionary.org/wiki/quintile) (5), [sextile](https://en.wiktionary.org/wiki/sextile) (6), [septile](https://en.wiktionary.org/wiki/septile) (7), [octile](https://en.wiktionary.org/wiki/octile) (8), [decile](https://en.wiktionary.org/wiki/decile) (10), [hexadecile](https://en.wiktionary.org/wiki/hexadecile) (16), [ventile](https://en.wiktionary.org/wiki/ventile)/[vigintile](https://en.wiktionary.org/wiki/vigintile) (20), [centile](https://en.wiktionary.org/wiki/centile)/[percentile](https://en.wiktionary.org/wiki/percentile) (100)